### PR TITLE
Taxonomy database

### PR DESCRIPTION
**Pull request by giosce**
Scripts to create MySQL database to hold taxonomy data.
Scripts to populate the database from the taxonomy toml files.

### CODE REVIEW

**Review by themightychris — CHANGES_REQUESTED** [review]
rename `database/` to `tools/mysql-loader`

File: database/brigade-index-and-taxonomy.sql [hunk 1]
```diff
@@ -0,0 +1,294 @@
...
+
+CREATE TABLE IF NOT EXISTS `projects_topics` (
+  `id` int(11) NOT NULL AUTO_INCREMENT,
+  `project_id` int(11) DEFAULT NULL,
+  `topic_id` int(11) DEFAULT NULL,
+  KEY `id` (`id`,`project_id`,`topic_id`)
+) ENGINE=InnoDB AUTO_INCREMENT=5394 DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_unicode_ci;
+
+
+CREATE TABLE IF NOT EXISTS `projects_topics_copy` (
```
Comment (themightychris) [inline database/brigade-index-and-taxonomy.sql:113]: delete this?

File: database/brigade-index-and-taxonomy.sql [hunk 2]
```diff
@@ -0,0 +1,294 @@
...
+-- Creating temporary table to overcome VIEW dependency errors
+CREATE TABLE `taxonomy_view` (
+	`synonym_2` VARCHAR(70) NULL COLLATE 'utf8mb4_unicode_ci',
+	`synonym` VARCHAR(70) NULL COLLATE 'utf8mb4_unicode_ci',
+	`count` BIGINT(21) NOT NULL
+) ENGINE=MyISAM;
+
+
+-- Creating temporary table to overcome VIEW dependency errors
+CREATE TABLE `taxonomy_view_2` (
```
Comment (themightychris) [inline database/brigade-index-and-taxonomy.sql:234]: delete?

File: database/proj-toml-2-mysql.py [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+              # first check if already exist
+              sql = "select id from topics where topic = %s"
+              cursor.execute(sql, ([topic]))
+              topic_id = cursor.fetchone()
+              if(topic_id == None):
+                sql = "INSERT INTO `topics_2` (`topic`) VALUES (%s)" 
+                cursor.execute(sql, ([topic]))
+                topic_id = cursor.lastrowid
+              sql = "INSERT INTO `projects_topics_2` (`project_id`,`topic_id`) VALUES (%s, %s)" 
+              cursor.execute(sql, (project_id, topic_id))
```
Comment (themightychris) [inline database/proj-toml-2-mysql.py:96]: should these `_2`s be here?

Comment (giosce) [reply database/proj-toml-2-mysql.py]: Removed _2 tables/statements.

File: database/brigade-index-and-taxonomy.sql (hunk 2 above)
Comment (giosce) [reply database/brigade-index-and-taxonomy.sql]: Removed the temporary tables, but I still have taxonomy_view_2, I don't know a much better name.

File: database/brigade-index-and-taxonomy.sql (hunk 1 above)
Comment (giosce) [reply database/brigade-index-and-taxonomy.sql]: Removed creation of _copy table.

**Review by themightychris — APPROVED** [review]

**Pull request merged**